### PR TITLE
Update matrix

### PR DESCRIPTION
**Pull request by rikardn**
Add new matrix methods `elementwise_mul`, `conjugate` and `conjugate_transpose`.

Also add is_zero as a property for the `Basic` class using the symengine `is_zero` function. Handling `tribool` by letting `tribool::indeterminate`, aka -1, become `None`

Building on #326

### CODE REVIEW

File: symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -3487,6 +3523,12 @@ cdef class DenseMatrixBase(MatrixBase):
         deref(self.thisptr).mul_matrix(deref(A_.thisptr), deref(result.thisptr))
         return result
 
+    def elementwise_mul_matrix(self, A):
```
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx:3526]: Use sympy name `multiply_elementwise`?

**Review by isuruf — COMMENTED** [review]
Looks good to me. Thanks for the PR

File: symengine/lib/symengine_wrapper.pyx (hunk 1 above)
Comment (rikardn) [reply symengine/lib/symengine_wrapper.pyx]: Ok, I have changed the name.

File: symengine/lib/symengine_wrapper.pyx [hunk 2]
```diff
@@ -3504,6 +3546,18 @@ cdef class DenseMatrixBase(MatrixBase):
...
+        cdef DenseMatrixBase result = self.__class__(self.nrows(), self.ncols())
+        deref(self.thisptr).conjugate(deref(result.thisptr))
+        return result
+
+    def conjugate_transpose(self):
+        cdef DenseMatrixBase result = self.__class__(self.nrows(), self.ncols())
+        deref(self.thisptr).conjugate_transpose(deref(result.thisptr))
+        return result
+
+
```
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx:3559]: To match sympy
```suggestion
    @property
    def H(self):
        return self.conjugate_transpose()
```

File: symengine/__init__.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
     LessThan, StrictGreaterThan, StrictLessThan, Eq, Ne, Ge, Le,
     Gt, Lt, And, Or, Not, Nand, Nor, Xor, Xnor, perfect_power, integer_nthroot,
     isprime, sqrt_mod, Expr, cse, count_ops, ccode, Piecewise, Contains, Interval, FiniteSet,
-    EmptySet, linsolve,
+    EmptySet, UniversalSet, Reals, Integers, linsolve,
```
Comment (isuruf) [inline symengine/__init__.py:15]: In sympy, the top level `Reals` is an object

**Comment by isuruf** [comment]
`Reals`, `Integers`, `UniversalSet` should be added in the `Singleton` class and when returning the class, we should return the object in the Singleton class. See `Half` for an example

**Comment by rikardn** [comment]
I put the EmptySet, UniversalSet, Reals and Integers sets into the Singletons class.

File: symengine/__init__.py (hunk 3 above)
Comment (isuruf) [reply symengine/__init__.py:15]:
```suggestion
    S.EmptySet as EmptySet, S.UniversalSet as UniversalSet, S.Reals as Reals, S.Integers as Integers, linsolve,
```

**Comment by rikardn** [comment]
Now I believe that the singletons are used correctly. I also added the rest of the numbers and matrix tests and the trace.

**Comment by isuruf** [comment]
thanks

**Pull request merged**